File: integrations/src/main/java/com/nivasafinance/integrations/framework/core/entity/ThirdPartyProviderConfig.java
package com.nivasafinance.integrations.framework.core.entity;

import com.nivasafinance.common.audit.AuditableEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "n_third_party_provider_config")
@NoArgsConstructor
@AllArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
public class ThirdPartyProviderConfig extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "provider")
    private String provider;

    @Column(name = "configs")
    private String configs;

    @Column(name = "active")
    private Boolean active;
}

File: integrations/src/main/kotlin/framework/core/entity/ThirdPartyProviderConfig.kt
package framework.core.entity

import audit.AuditableEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.UUID
import org.javers.core.metamodel.annotation.Entity as JaversEntity

@Table(name = "third_party_provider_config")
@Entity
@JaversEntity
data class ThirdPartyProviderConfig(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID,

    @Column(name = "name")
    val name: String,

    @Column(name = "provider")
    val provider: String,

    @Column(name = "configs")
    val configs: String,

    @Column(name = "active")
    val active: Boolean,

) : AuditableEntity()
